crates/rpc/src/handlers/state.rs

```diff
@@ -77,3 +78,46 @@ pub async fn get_state_root(state_id: ID, db: ReamDB) -> Result<impl Reply, Reje
         StatusCode::OK,
     ))
 }
+#[derive(Debug, Serialize, Deserialize, Default)]
+pub struct WithdrawalData {
+    #[serde(with = "serde_utils::quoted_u64")]
+    validator_index: u64,
+    #[serde(with = "serde_utils::quoted_u64")]
+    amount: u64,
+    #[serde(with = "serde_utils::quoted_u64")]
+    withdrawable_epoch: u64,
+}
+// Called by `/states/{state_id}/get_pending_partial_withdrawals` to get pending partial withdrawals
+// for state with given stateId
+pub async fn get_pending_partial_withdrawals(
+    state_id: ID,
+    db: ReamDB,
+) -> Result<impl Reply, Rejection> {
+    let state = get_state_from_id(state_id, &db).await?;
+
+    let withdrawals = state.get_expected_withdrawals();
+    let partial_withdrawals: Vec<Withdrawal> = withdrawals
+        .into_iter()
+        .filter(|w| {
```

Comment:
```suggestion
        .filter(|withdrawal| {
```
can we not use 1 letter variable names